densecatfeature.go

```diff
@@ -871,7 +870,6 @@ func (f *DenseCatFeature) BestCatSplitBig(target Target, cases *[]int, parentImp
 		nPartitions.Lsh(nPartitions, uint(maxEx-2))
 		maxPart = big.NewInt(2)
 		maxPart.Lsh(maxPart, uint(nCats-2))
-		randgn = rand.New(rand.NewSource(0))
```

Comment:
This used to generated the same numbers every time due to the static seed. I assume that's not correct, so I changed it to use the rand source from BestSplitAllocs. But this means the test that relies on same results every time is now broken (the results are the same every time now too, but that's because rand.Seed() is not called in tests - if you call it, the results are different every time).